docs/demos/demos/intro_demo.py

```diff
@@ -0,0 +1,256 @@
+# ## Simulating a circuit
+#
+# `dwave-gate` comes with a performant state-vector simulator. It can be called by passing a circuit to the `simulate` method, which will update the quatum state stored in the circuit, accessible via `Circuit.state`.
+
+from dwave.gate.simulator import simulate
+
+###############################################################################
+# We create a circuit object with 2 qubits and 1 bit in a quantum and classical registers respectively --- the bit is required to store a single qubit measurement --- and then apply a Hadamard gate and a CNOT gate to the circuit.
+
+circuit = Circuit(2, 1)
+
+with circuit.context as (q, c):
+    ops.Hadamard(q[0])
+    ops.CNOT(q[0], q[1])
+
+###############################################################################
+# We can now simulate the circuit, which will update its stored quantum state.
+
+simulate(circuit)
+
+###############################################################################
+# Printing the state reveals the expected state-vector $\frac{1}{\sqrt{2}}\left[1,
+# 0, 0, 1\right]$ corresponding to the state:
+#
+# $$\vert\psi\rangle = \frac{1}{\sqrt{2}}\left(\vert00\rangle + \vert11\rangle\right)$$
+
+circuit.state
+
+###############################################################################
+# ## Measurements
+#
+# Measurements work like any other operation in dwave-gate. The main difference is that the operation generates a measurement value when simulated which can be stored in the classical register by piping it into a classical bit.
+#
+# We can reuse the circuit from above by simply unlocking it and appending a `Measurement` to it.
+
+circuit.unlock()
+with circuit.context as (q, c):
+    m = ops.Measurement(q[1]) | c[0]
+
+###############################################################################
+# :::note
+# We stored the measurement instance as `m`, which we can use for post-processing. It's also possible to do this with all other operations in the same way, allowing for multiple identical operation applications.
+# ```python
+# with circuit.context as q, _:
+#     single_x_op = ops.X(q[0])
+#     # apply the X-gate again to the second qubit
+#     # using the previously stored operation
+#     single_x_op(q[1])
+# ```
+# This procedure can also be shortened into a single line for further convience.
+# ```python
+# ops.CNOT(q[0], q[1])(q[1], q[2])(q[2], q[3])
+# ```
+# :::
+
+###############################################################################
+# The circuit should now contain 3 operations: a Hadamard, a CNOT and a measurment.
+
+print(circuit)
+
+###############################################################################
+# When simulating this circuit, the measurement will be applied and the measured value will be stored in the classical register. Since a measurement will affect the quantum state, the resulting state will have collapsed into the expected result dependent on the value which has been measured.
```

Comment:
```suggestion
# When you simulate this circuit, the measurement is applied and the measured 
# value is stored in the classical register. Since a measurement affects the 
# quantum state, the resulting state has collapsed into the expected result 
# dependent on the value which has been measured.
```

Another dangler: "When simulating this circuit, the measurement" has the measurement simulating the circuit